creditTest1/task1/encoder/Encoder.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace encoder;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Class for encoding and decoding sequences of bytes.
+/// </summary>
+public static class Encoder
+{
+    /// <summary>
+    /// Encodes sequence of bytes.
+    /// </summary>
+    /// <param name="data">data to encode.</param>
+    /// <returns>Encoded data and ratio of length of encoded data to source data.</returns>
+    public static (List<byte> encodedData, double compressionRatio) Encode(IEnumerable<byte> data)
+    {
+        int sourceLength = 0;
+        int encodedDataLength = 0;
+        List<byte> encodedData = data.Aggregate(new List<byte>(), (List<byte> list, byte @byte) =>
+        {
+            sourceLength++;
+            if (list.Count > 0 && list.Last() == @byte && list[list.Count - 2] < byte.MaxValue)
+            {
+                list[list.Count - 2]++;
+            }
+            else
+            {
+                list.Add(1);
+                list.Add(@byte);
+                encodedDataLength += 2;
+            }
+
+            return list;
+        });
+
+        return (encodedData, encodedDataLength / (double)sourceLength);
+    }
+
+    /// <summary>
+    /// Decodes sequence of bytes.
+    /// </summary>
+    /// <param name="source">data to decode.</param>
+    /// <returns>Decoded data.</returns>
+    public static List<byte> Decode(IEnumerable<byte> source)
+        => AggregateTwo(source, new List<byte>(), (List<byte> list, (byte amount, byte x) v) =>
+        {
+            list.AddRange(Enumerable.Range(0, v.amount).Select(_ => v.x));
+            return list;
+        });
+
+    private static TOut AggregateTwo<TIn, TOut>(IEnumerable<TIn> source, TOut seed, System.Func<TOut, (TIn, TIn), TOut> func)
```

Comment:
```suggestion
    public static List<byte> Decode(IEnumerable<byte> source)
        => sourse.AggregateTwo(new List<byte>(), (List<byte> list, (byte amount, byte x) v) =>
        {
            list.AddRange(Enumerable.Range(0, v.amount).Select(_ => v.x));
            return list;
        });

    private static TOut AggregateTwo<TIn, TOut>(this IEnumerable<TIn> source, TOut seed, System.Func<TOut, (TIn, TIn), TOut> func)
```
Просто к сведению.
Метод `AggregateTwo` фактически является расширением интерфейса `IEnumerable`.
И это можно явно выразить с помощью ключевого слова `this` перед первым параметром.
В таком случае можно будет этот метод вызывать в ООП-стиле.